packages/expo-task-manager/src/TaskManager.ts

```diff
@@ -91,7 +92,19 @@ const tasks: Map<string, TaskManagerTaskExecutor<any>> = new Map<
   TaskManagerTaskExecutor<any>
 >();
 
-function _validateTaskName(taskName) {
+function _isExpoGo(): boolean {
+  return typeof expo !== 'undefined' && !!globalThis.expo?.modules?.ExpoGo;
+}
+
+function _validate(taskName: unknown) {
+  if (_isExpoGo()) {
+    const message =
+      '`TaskManager` functionality is limited in Expo Go:\n' +
+      'On Android, it is not available at all.\n' +
+      'On iOS, it is limited to foreground execution.\n' +
+      'Please use a development build to avoid limitations. Learn more: https://expo.fyi/dev-client.';
+    console.warn(message);
```

Comment:
One thing that we're usually doing with warnings is to output them only once, so we don't clutter user's console. See 👇 as an example:
https://github.com/expo/expo/blob/b1408b3f6c73462933f5b07991d074a2a3dc2702/packages/expo-barcode-scanner/src/BarCodeScanner.tsx#L157-L162